com.unity.multiplayer.mlapi/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -578,8 +585,16 @@ private void ProcessNetworkBehaviour(TypeDefinition typeDefinition)
                     instructions.Add(processor.Create(OpCodes.Ldftn, method));
                     instructions.Add(processor.Create(OpCodes.Newobj, m_NetworkHandlerDelegateCtor_MethodRef));
                     instructions.Add(processor.Create(OpCodes.Call, m_NetworkManager_ntable_Add_MethodRef));
+
+                    // NetworkManager.__rpc_name_table.Add(HandlerHash, RpcMethodName);
+                    instructions.Add(processor.Create(OpCodes.Ldsfld, m_NetworkManager_rpc_name_table_FieldRef));
+                    instructions.Add(processor.Create(OpCodes.Ldc_I4, unchecked((int)hash)));
+                    instructions.Add(processor.Create(OpCodes.Ldstr, method.Name.Substring(0, method.Name.Length-"__nhandler".Length)));
```

Comment:
instead of doing this grab xxx__handler and cut __handler part trick here, we could grab the name from `methodDefinition` + `methodDefHash` above and inject outside this foreach block with another foreach iterating over `rpcMethodHash`-`rpcMethodName` combo.
I can reach out later next week to refactor this together.

Reply:
I tried something like that, but I couldn't get it to work for some reason. I'm all open for better solution.